sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGetJsonObject.scala

```diff
@@ -32,8 +32,20 @@ case class GpuGetJsonObject(json: Expression, path: Expression)
   override def nullable: Boolean = true
   override def prettyName: String = "get_json_object"
 
-  override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {    
-    lhs.getBase().getJSONObject(rhs.getBase, 
+  def normalizeJsonPath(path: GpuScalar): Scalar = {
+    val pathStr = if (path.isValid) {
+      path.getValue.toString()
+    } else {
+      throw new IllegalArgumentException("Invalid path")
+    }
+    // Remove all leading whitespaces before names, so we erase all whitespaces after . or ['
+    val normalizedPath = pathStr.replaceAll("""\.(\s+)""", ".").replaceAll("""\['(\s+)""", "['")
```

Comment:
I don't like us using regular expressions here. It might work, but I don't see it as a good long term solution. Could we please take the code that Spark uses to parse a JSON path.

https://github.com/apache/spark/blob/78f7c30e140fd8cf4a80b783dd7e9ee4d1b4d7e2/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala#L59

And modify it so that we output a normalized JSON path instead.  That would also give us the ability to detect errors in the path before we send it to CUDF and make sure that we are compatible with what CUDF supports.

Meaning we really just reuse their code and then convert the `Option[List[PathInstruction]]` back to a string.

```
parsedPath match {
  case None => // ERROR return a column of nulls
  case Some(lp) =>
    val ret = new StringBuffer()
    ret.append('$')
    lp.foreach {
      case Subscript | Key => // Ignored we don't care about these
      case Wildcard => ret.append("[*]")
      case Named(name) => ret.append(s"['${name}']")
      case Index(idx) => ret.append(s"[$idx]")
      case _ => ERROR!!!!
    }
}
```

Reply:
Thanks! Updated to use JsonPathParser, this approach also looks to fix some other issue related to json path.

Reply:
Updated the PR description to explain why this PR can fix these issues.